back/src/route/userController.js

```diff
@@ -52,4 +54,28 @@ module.exports = class UserController{
             return res.status(400).json({"success": false, "error": exception});
         });
     }
+
+    async edit(req, res){
+        const {currentEmail, newEmail, newPassword, newUserName} = req.body;
+        let encryptedNewPassword;
+        if (newPassword != "") {
```

Comment:
We need to check if password is only white space too. You can do something like if newPassword.trim() != "" (trim removes white space).

We also need to check that the newEmail is a valid email and we need to check that newUserName is not white space only (if newUserName.trim() != "")